app/api/groups/get_current_user_team_by_item.go

```diff
@@ -0,0 +1,65 @@
+package groups
+
+import (
+	"errors"
+	"net/http"
+	"strconv"
+
+	"github.com/go-chi/render"
+	"github.com/jinzhu/gorm"
+
+	"github.com/France-ioi/AlgoreaBackend/app/service"
+)
+
+// swagger:operation GET /current-user/teams/by-item/{item_id} users groups teamGetByItemID
+// ---
+// summary: Get current user's team for item
+// description: >
+//   The team identified by `item_id` i.e. a group which:
+//
+//     * has `idTeamItem` equal to the input `item_id`,
+//
+//     * is a direct parent (i.e. via `groups_groups` with `sType` = "invitationAccepted"/"requestAccepted"/"joinedByCode")
+//       of the authenticated user’s `selfGroup`,
+//
+//     * is of type "Team".
+//
+//
+//   If there are several matching teams, returns the first one in the order of `groups.ID`.
+// parameters:
+// - name: item_id
+//   type: integer
+//   required: true
+//   in: path
+// responses:
+//   "200":
+//     description: OK. Success response with the team's ID
+//     schema:
+//       type: object
+//       properties:
+//         group_id:
+//           type: integer
+//       required: [group_id]
+//   "401":
+//     "$ref": "#/responses/unauthorizedResponse"
+//   "404":
+//     "$ref": "#/responses/notFoundResponse"
+//   "500":
+//     "$ref": "#/responses/internalErrorResponse"
+func (srv *Service) getCurrentUserTeamByItem(w http.ResponseWriter, r *http.Request) service.APIError {
+	itemID, err := service.ResolveURLQueryPathInt64Field(r, "item_id")
+	if err != nil {
+		return service.ErrInvalidRequest(err)
+	}
+
+	var teamID int64
+	user := srv.GetUser(r)
+	err = srv.Store.Groups().TeamGroupByTeamItemAndUser(itemID, user).PluckFirst("groups.ID", &teamID).Error()
```

Comment:
The function is actually a bit misleading as one may think it is a "GroupBy" ;-). Any better proposal is welcome. 🙂

Reply:
'Items()', 'Groups()', 'GroupGroups()', 'Users()'.... Do you want me to rename them into 'ItemStore()', 'GroupStore()', 'GroupGroupStore()', 'UserStore()' and so on?

Reply:
No, it is not not what I mean. In "TeamGroupByTeamItemAndUser" can be read as "Team GroupBy TeamItem And User" while it is actually "TeamGroup (filtered) By Team Item And User".  I was first confused by this groupby when re-reading it today, that why I mention it.

Reply:
‘By’ -> ‘For’ then?

Reply:
Yes, for instance. Or "FilterBy" bit that's maybe long.

Reply:
fixed